controllers/helmrelease_controller.go

```diff
@@ -267,14 +265,6 @@ func (r *HelmReleaseReconciler) reconcileChart(ctx context.Context, hr *v2.HelmR
 }
 
 func (r *HelmReleaseReconciler) release(ctx context.Context, log logr.Logger, hr v2.HelmRelease, source sourcev1.Source, values chartutil.Values) (v2.HelmRelease, error) {
-	// Acquire lock
```

Comment:
@seaneagan can you please double check if my observations are right?

Reply:
Yes it seems controller-runtime ensures only 1 concurrent reconcile per object and `observedGeneration` etc get updated synchronously inside each reconciliation, so makes sense to me.